src/automata/ChannelRequester.cpp

```diff
@@ -130,15 +130,15 @@ void ChannelRequester::endStream(StreamCompletionSignal signal) {
   ConsumerBase::endStream(signal);
 }
 
-void ChannelRequester::onNextFrame(Frame_PAYLOAD&& frame) {
+void ChannelRequester::handlePayload(Payload&& payload, bool complete, bool flagsNext) {
```

Comment:
bool arguments is kinda horrible, why exactly are we no longer passing the frame? if we really don't want to pass the frame could we create some sort of Flags type that allows easier querying? `flags.has(FrameFlags::COMPLETE)`?